File: app/src/main/java/com/chinachino/mvvm/Repositories/MovieDetailsRepository.java
package com.chinachino.mvvm.Repositories;

import androidx.lifecycle.LiveData;

import com.chinachino.mvvm.Requests.MovieAPIClient;
import com.chinachino.mvvm.models.Details;
//
public class MovieDetailsRepository {
    private static MovieDetailsRepository instance;
    private MovieAPIClient movieAPIClient;

    public static MovieDetailsRepository getInstance() {
        if (instance == null) {
            instance = new MovieDetailsRepository();
        }
        return instance;
    }

    public LiveData<Boolean> isRequestTimeOut(){
        return movieAPIClient.isRequestTimeOut();
    }

    private MovieDetailsRepository() {
        movieAPIClient = MovieAPIClient.getInstance();
    }


    public LiveData<Details> getMovieDetails(){
        return movieAPIClient.getMovieDetails();
    }

    public void SearchMovieDetails(int movieID){
        movieAPIClient.SearchMovieID(movieID);
    }
}

File: app/src/main/java/com/chinachino/mvvm/repositories/MovieDetailsRepository.kt
package com.chinachino.mvvm.repositories

import androidx.lifecycle.LiveData
import com.chinachino.mvvm.requests.MovieAPIClient
import com.chinachino.mvvm.models.Details

//
class MovieDetailsRepository private constructor() {
    private val movieAPIClient: MovieAPIClient = MovieAPIClient()
    val isRequestTimeOut: LiveData<Boolean>
        get() = movieAPIClient.isRequestTimeOut()
    val movieDetails: LiveData<Details?>
        get() = movieAPIClient.movieDetails

    fun searchMovieDetails(movieID: Int) {
        movieAPIClient.searchMovieID(movieID)
    }

    companion object {
        var instance: MovieDetailsRepository? = null
            get() {
                if (field == null) {
                    field = MovieDetailsRepository()
                }
                return field
            }
            private set
    }

}